Shelf.cs

```diff
@@ -0,0 +1,158 @@
+            }
+        }
+        public int ShelfSpace 
+        {
+            get { return shelfSpace; } 
+            set {
+                if (value > 0)
+                    shelfSpace = value;
+                else
+                    Console.WriteLine("shelfSpace must be positive");
+            }
+        }
+        public List<Item> Items 
+        {
+            get { return items; } 
+            set {
+                if (!(value is null))
+                {
+                    items = value;
+                }
+                else
+                {
+                    items = new List<Item>();
+                }
+                ; } 
+        }
+
+        public Shelf(int floorNumber,int shelfSpace)
+        {
+           
+            this.ID = nextID++;
+            this.floorNumber = floorNumber;
+            this.shelfSpace = shelfSpace;
+            this.items = new List<Item>();
+            
+        }
+        public Shelf()
+        {
+            this.ID = nextID++;
+        }
+
+
+        public double spaceInShelf()
+        {
+            double wastedSpace = 0;
+           
+            foreach (Item item in items)
+            {
+
+                wastedSpace += item.SpaceNeed;
+
+            }
+            return  (ShelfSpace - wastedSpace);
+
+
+            
+        }
+
+        public Boolean FindItemInShelf(int itemId)
+        {
+            foreach(Item item in items)
+            {
+                if(item.Id==itemId)
+                    return true;
+            }
+            return false;
+        }
+
+        public Item RemoveItemFromShelf(int itemId)
+        {
+            Item removeItem=new Item();
+            foreach(Item item in items)
+            {
+                if (item.Id == itemId)
+                {
+                    removeItem.Id = item.Id;
+                    removeItem.ShelfId = item.ShelfId;
+                    removeItem.SpaceNeed= item.SpaceNeed;
+                    removeItem.Name= item.Name;
+                    removeItem.ExpiryDate= item.ExpiryDate;
+                    removeItem.Type= item.Type;
+                    removeItem.Kashroot= item.Kashroot;
+
+                    //removeItem = item;//לבדוקקקקק
+                    items.Remove(item);
+                    break;
+                }
+            }
+            return removeItem;
+        }
+
+        public void addItemToShelf(Item item)
+        {
+            this.items.Add(item);
+            item.ShelfId= this.ID;//עדכון המדף אליו הוא נכנס
+            Console.WriteLine("The item has been placed on the shelf {0}", ID);
+        }
+
+        public void cleanTheShelf()
+        {
+            DateTime currentDate = DateTime.Now;
+            for (int i = 0; i < Items.Count; i++)
+            {
+
+                if (Items[i].ExpiryDate.Date < currentDate.Date)
+                {
+
+                    Items.Remove(Items[i]);
+                }
+            }
```

Comment:
What if we have 2 object and you remove the first one? will you check the second one? think about it 🧠